Gems/Atom/Feature/Common/Code/Source/Decals/DecalTextureArray.cpp

```diff
@@ -298,10 +298,17 @@ namespace AZ
             {
                 return {};
             }
-            AZ_Assert(
-                mip < image->GetImageDescriptor().m_mipLevels,
-                "It is expected that all decals in a texture array must have the same number of mips which may not be the case here. "
-                "Please ensure that all the materials within m_materials are pointing to textures with same mips.");
+            // Gruber patch begin // STA : detailed error log and exit to prevent crash
+            if (mip >= image->GetImageDescriptor().m_mipLevels)
+            {
+                AZ_Error("DecalTextureArray", false,
+                    "It is expected that all decals in a texture array must have the same number of mips which may not be the case here. "
+                    "Please ensure that all the materials within m_materials are pointing to textures with same mips.\n"
+                    "Needed mip #%i, max mip #%i, asset '%s'."
+                    , mip, image->GetImageDescriptor().m_mipLevels - 1, m_materials[arrayLevel].m_materialAssetData.GetHint().c_str());
+                return {};
+            }
+            // Gruber patch end // STA
```

Comment:
Replace the gruber patch w/ #if defined(CARBONATED) / etc.

This will make future merges and PR back to main engine easier.

Reply:
@carb-lloyd
In this particular case there is obvious deficiency of missing `return {};` regardless of whether this is a Catbonated project or not.
Will not 
#if defined(CARBONATED) ... #endif
mislead that this is a Catbonated project specific patch?

Maybe we need another tag for bug-fixing which later should be updated to OpenSource?

Reply:
@Aleks-Starykh-carb  - The exact same can be said about "gruber patch" ;)
All changes to the engine that _we_ make, get to be guarded with #if defined(CARBONATED).

When we get to a good point to start making PRs back to the engine, we will remove the guards and create PRs without them.  Same with the gruber patches - we'd have to remove them before making the PRs.  We prefer the #if defs.

Reply:
Well, most Gruber patches to O3DE were made to support legacy LY features, and were reasonably guarded with `#ifdef CARBONATED`.

Reply:
> Well, most Gruber patches to O3DE were made to support legacy LY features, and were reasonably guarded with `#ifdef CARBONATED`.

And they should all be changed to #if defined(CARBONATED) instead.  

In fact, I have a ticket available that the engine must be able to be build and compile without CARBONATED being specified.

Reply:
Yes, sure O3DE should compile without patches supporting legacy LY features.
I not fixing native O3DE bugs another case?